lib/src/components/NxNavigationSidebar/NxNavigationSidebar.scss

```diff
@@ -0,0 +1,160 @@
+/**
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+@import '../../scss-shared/nx-variables';
+@import '../../scss-shared/nx-text-helpers';
+@import '../../scss-shared/nx-container-helpers';
+
+$nx-sidebar-width__open: 255px;
+$nx-sidebar-width__closed: 72px;
+$nx-sidebar-logo-height: 34px;
+$nx-sidebar-link-height: 40px;
+
+$nx-sidebar-text-color: $nx-white;
+$nx-sidebar-background-color: $nx-blue-1000;
+$nx-sidebar-active-color: $nx-indigo-400;
+$nx-sidebar-hover-color: $nx-indigo-600;
+$nx-sidebar-selected-color: $nx-indigo-800;
+$nx-sidebar-separator-color: $nx-grey-900;
+
+.nx-page-sidebar--navigation {
+  @include container-vertical;
+
+  background-color: $nx-sidebar-background-color;
+  color: $nx-sidebar-text-color;
+  display: flex;
+  flex: none;
+  flex-direction: column;
+  width: $nx-sidebar-width__open;
+  padding: $nx-spacing-l 0 $nx-spacing-xxl 0;
+}
+
+.nx-sidebar__header {
+  @include container-vertical;
+
+  display: flex;
+  flex-direction: column;
+  margin: 0 $nx-spacing-l;
+}
+
+.nx-sidebar__toggle {
+  align-self: flex-end;
+  border-radius: 50%;
+  color: $nx-sidebar-text-color;
+  height: 41px;
+  margin: 0;
+  width: 41px;
+
+  &:hover, &:focus {
+    background-color: $nx-sidebar-hover-color;
+    border-color: $nx-sidebar-hover-color;
+  }
+
+  &:active {
+    background-color: $nx-sidebar-active-color;
+    border-color: $nx-sidebar-active-color;
+  }
+}
+
+.nx-sidebar__product-info {
+  @include container-horizontal;
+
+  align-items: center;
+  display: flex;
+  color: $nx-sidebar-text-color;
+  text-decoration: none;
+}
+
+.nx-sidebar__logo {
+  height: $nx-sidebar-logo-height;
+  margin-right: $nx-spacing-xs;
+}
+
+.nx-sidebar__content {
+  flex: 1 0 auto;
+  font-size: $nx-font-size-xs;
+  margin: $nx-spacing-l 0 0;
+
+  a {
+    color: $nx-sidebar-text-color;
+  }
+}
+
+.nx-sidebar__navigation {
```

Comment:
Just realized, a bunch of these are `.nx-sidebar...` but the parent Block is `.nx-page-sidebar`.  That said, I'm not sure I'd be thrilled about having `.nx-page-sidebar--navigation` and `.nx-page-sidebar__navigation` as two separate things.  Maybe this one should be `.nx-page-sidebar__nav` or some other minor but noticeable difference

Reply:
> Just realized, a bunch of these are `.nx-sidebar...` but the parent Block is `.nx-page-sidebar`. That said, I'm not sure I'd be thrilled about having `.nx-page-sidebar--navigation` and `.nx-page-sidebar__navigation` as two separate things. Maybe this one should be `.nx-page-sidebar__nav` or some other minor but noticeable difference

I suck at naming — Tell me what to name it and I'll refactor it

Reply:
> I suck at naming — Tell me what to name it and I'll refactor it

Well all of the `.nx-sidebar__*` need to be `.nx-page-sidebar__*`.  Beyond that, as I said above my suggestion for this element is `.nx-page-sidebar__nav`.  @glenhunter do you have any thoughts?

Reply:
> > I suck at naming — Tell me what to name it and I'll refactor it
> 
> Well all of the `.nx-sidebar__*` need to be `.nx-page-sidebar__*`. Beyond that, as I said above my suggestion for this element is `.nx-page-sidebar__nav`. @glenhunter do you have any thoughts?

@raveneyex and I have gone around and around on this. I think that it's a mistake to base the styles on `nx-page-sidebar`. there is so little overlap between the two that the risks of changes to `.nx-page-sidebar` effecting this in the future outweigh the very minimal code sharing.

For now I suggested that he replace `nx-page-sidebar--navigation` with `nx-page-sidebar-operable` so that he can move forward, but I'm not at all happy with that name.

Reply:
Also FWIW I'd be inclined to make this and the `content` sections below into blocks rather than elements to avoid the tortured element of element (`nx-sidebar__navigation__link`) class names.

Reply:
@glenhunter @rpokorny I ended up renaming stuff in 87be8aa113109e6591853229f3c6727466594dd1.

I used `.nx-page-sidebar-operable` for the main class and `nx-sidebar-navigation` for the nav section. 
I do not love them and will not defend them and will change them if anyone has a better recommendation.

Reply:
> I used .nx-page-sidebar-operable for the main class and .nx-sidebar-navigation for the nav section.
I do not love them and will not defend them and will change them if anyone has a better recommendation.

Personally I still liked it being a modifier of `.nx-page-sidebar`.  There is a fair amount of overlap: if you go back to commit ca228d54 you'll see that `.nx-page-sidebar--navigation` is only overriding 3 out of the 8 properties set my `.nx-page-sidebar`.  

I do however like the idea of making the nav section a block instead of an element.

> `nx-sidebar__navigation__link`

FWIW it doesn't have to be like that; we've avoided doubled-up element syntax like that in the past.  It would have just been `.nx-sidebar__navigation-link`.

Reply:
I really need both of you @glenhunter and @rpokorny to decide on the naming. I'm tired of this naming back and forth.

Reply:
> I really need both of you @glenhunter and @rpokorny to decide on the naming. I'm tired of this naming back and forth.

Using agreed upon name in 707bbe8eed16f8271ce966c4480d4c6c033992a1

Reply:
> Using agreed upon name in 707bbe8

React component names should be updated to match

Reply:
> > Using agreed upon name in [707bbe8](https://github.com/sonatype/sonatype-react-shared-components/commit/707bbe8eed16f8271ce966c4480d4c6c033992a1)
> 
> React component names should be updated to match

Done in 963608412d19c847eacaa42ce8b5479a86bb2b7e